staging/src/k8s.io/apiserver/pkg/authentication/token/tokenfile/tokenfile.go

```diff
@@ -88,10 +90,10 @@ func NewCSV(path string) (*TokenAuthenticator, error) {
 	}, nil
 }
 
-func (a *TokenAuthenticator) AuthenticateToken(value string) (user.Info, bool, error) {
+func (a *TokenAuthenticator) AuthenticateToken(ctx context.Context, value string) (*authenticator.Response, bool, error) {
```

Comment:
TODO to take implicit audiences at TokenAuthenticator construction time, and if ctx is audience-limited, ensure it intersects with the implicit audiences, and echo an audience in the response